lib/ibm_cloud_sdk_core/utils.rb

```diff
@@ -63,10 +63,20 @@ def load_from_vcap_services(service_name)
   vcap_services = ENV["VCAP_SERVICES"]
   unless vcap_services.nil?
     services = JSON.parse(vcap_services)
+    credentials = ""
+    # search for matching inner name value
+    services.each do |_key, val|
+      service = val.detect { |item| item["name"] == service_name }
+      credentials = service["credentials"] unless service.nil?
+      break unless credentials.nil? || credentials.empty?
+    end
     config = {}
-    credentials = services[service_name][0]["credentials"] if services.key?(service_name)
-    return config if credentials.nil?
-
+    # if no matching inner key is found, then search outer keys
+    if credentials.nil? || credentials.empty?
+      credentials = services[service_name][0]["credentials"] if services.key?(service_name) && !services[service_name].empty?
+      return config if credentials.nil? || credentials.empty?
+    end
+    # store credentials
     credentials.each do |key, val|
       config.store(key.to_sym, val)
     end
```

Comment:
Do we need to explicitly return "config" here?   I know Ruby has some psychodelic behavior re: return values, so just asking :)

Reply:
This is one of those "this was already there and it worked" kind of things 😄, but after doing some reading looks like rudy allows you to also do explicit returns within a method so I think this is correct.